flang/lib/Optimizer/Analysis/AliasAnalysis.cpp

```diff
@@ -319,6 +330,32 @@ AliasAnalysis::Source AliasAnalysis::getSource(mlir::Value v,
           breakFromLoop = true;
         })
         .Case<hlfir::DeclareOp, fir::DeclareOp>([&](auto op) {
+          // If declare operation is inside omp target region,
+          // continue alias analysis outside the target region
+          if (llvm::isa<omp::TargetOp>(op->getParentOp())) {
+            omp::TargetOp targetOp =
+                llvm::cast<omp::TargetOp>(op->getParentOp());
```

Comment:
nit: This can be written more efficiently as `if (auto targetOp = llvm::dyn_cast<omp::TargetOp>(op->getParentOp()))`.

Reply:
Fixed, thanks